### PR TITLE
Localize HelmChartInflationGenerator

### PR DESCRIPTION
**Pull request by annasong20**
This PR localizes the `HelmChartInflationGenerator` custom transformer. If chart home isn't specified, the default home is copied.

### CODE REVIEW

File: api/krusty/localizer/runner_test.go [hunk 1]
```diff
@@ -452,13 +452,11 @@ func TestResourcesRepoNotFile(t *testing.T) {
...
-	const readmeErr = `mapping values are not allowed in this context`
-	fileErr := fmt.Sprintf(`invalid resource at file "%s": MalformedYAMLError:`, repo)
+	fileErr := fmt.Sprintf(`invalid resource at file "%s": MalformedYAMLError`, repo)
 	rootErr := fmt.Sprintf(`unable to localize root "%s": unable to find one of 'kustomization.yaml', 'kustomization.yml' or 'Kustomization'`, repo)
 	var actualErr PathLocalizeError
 	require.ErrorAs(t, err, &actualErr)
 	require.Equal(t, repo, actualErr.Path)
-	require.ErrorContains(t, actualErr.FileError, readmeErr)
 	require.ErrorContains(t, actualErr.FileError, fileErr)
 	require.ErrorContains(t, actualErr.RootError, rootErr)
```
Comment (annasong20) [inline api/krusty/localizer/runner_test.go:461]: I removed the `readmeErr` that's specific to the contents of the kustomize repo README.md and irrelevant to our test, which just expects that the README is invalid yaml.

The over-specificity of this test previously failed the ci pipeline on master: https://github.com/kubernetes-sigs/kustomize/actions/runs/4018832421/jobs/6904935339

File: api/internal/localizer/localizer_test.go [hunk 2]
```diff
@@ -1461,6 +1461,62 @@ func TestCopyChartHomeError(t *testing.T) {
 	}
 }
 
+func TestLocalizeGeneratorsHelm(t *testing.T) {
+	files := map[string]string{
+		"kustomization.yaml": `generators:
+- default.yaml
```
Comment (natasha41575) [inline api/internal/localizer/localizer_test.go:1467]: Does `localize` currently handle generators defined inline? This is not required for alpha imo, but if it already handled, can we add a test? If not, can we add a TODO?

**Review by natasha41575 — APPROVED** [review]
lgtm, just a small comment above

File: api/internal/localizer/localizer_test.go (hunk 2 above)
Comment (annasong20) [reply api/internal/localizer/localizer_test.go:1518]: Done. Yes, we handle inline generators and I added a test.

**Review by natasha41575 — COMMENTED** [review]
/lgtm

**Pull request merged**